lm75bd/lm75bd.c

```diff
@@ -26,15 +29,47 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
   return ERR_CODE_SUCCESS;
 }
 
-error_code_t readTempLM75BD(uint8_t devAddr, float *temp) {
+error_code_t readTempLM75BD(uint8_t devAddr, float *temp)
+{
+
+  if (temp == NULL)
+    return ERR_CODE_INVALID_ARG;
+
+  uint8_t buffer[READ_BUF_SIZE] = {0};
+  uint8_t tempBuf = TEMP_ADD;
+
+  error_code_t sendErr = i2cSendTo(devAddr, &tempBuf, 1);
+  if (sendErr != ERR_CODE_SUCCESS)
+  {
+    return sendErr;
+  }
+
+  error_code_t readErr = i2cReceiveFrom(devAddr, &buffer, 2);
+  if (readErr != ERR_CODE_SUCCESS)
+  {
+    return readErr;
+  }
+
+  uint16_t tempBytes = ((buffer[0] << 8) | buffer[1]) >> 5;
+
+  if (!(buffer[0] & 0b10000000))
+  {
+    *temp = tempBytes * 0.125f;
+  }
+  else
+  {
+    *temp = -(((~tempBytes) + 1) & 0x07FF) * 0.125f;
+  }
```

Comment:
Note that this is a little more complicated than it needs to be. It could instead be done with:
```
int16_t regVal = buffer[0] << 8 | buffer[1];
*temp = (float)(regVal >> 5) * 0.125
```